stress/shell/src/main/java/alluxio/stress/cli/StressJobServiceBench.java

```diff
@@ -0,0 +1,280 @@
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Job Service stress bench.
+ */
+public class StressJobServiceBench extends Benchmark<JobServiceBenchTaskResult> {
+  private static final Logger LOG = LoggerFactory.getLogger(StressJobServiceBench.class);
+  public static final int MAX_RESPONSE_TIME_BUCKET_INDEX = 0;
+  @ParametersDelegate
+  private JobServiceBenchParameters mParameters = new JobServiceBenchParameters();
+
+  /**
+   * Creates instance.
+   */
+  public StressJobServiceBench() {}
+
+  /**
+   * @param args command-line arguments
+   */
+  public static void main(String[] args) {
+    mainInternal(args, new StressJobServiceBench());
+  }
+
+  @Override
+  public void prepare() throws Exception {
+    FileSystemContext fsContext =
+        FileSystemContext.create(new InstancedConfiguration(ConfigurationUtils.defaults()));
+    FileSystem fileSystem = FileSystem.Factory.create(fsContext);
+    long start = CommonUtils.getCurrentMs();
+    deleteBasePath(fileSystem);
+    long deleteEnd = CommonUtils.getCurrentMs();
+    LOG.info("Cleanup delete took: {} s", (deleteEnd - start) / 1000.0);
+    createFiles(fileSystem, mParameters.mNumFilesPerDir, mParameters.mNumDirs,
+        mParameters.mFileSize);
+    long createEnd = CommonUtils.getCurrentMs();
+    LOG.info("Create files took: {} s", (createEnd - deleteEnd) / 1000.0);
+  }
+
+  private void createFiles(FileSystem fs, int numFiles, int numDirs, int fileSize)
+      throws IOException, AlluxioException {
+    CreateFilePOptions options =
+        CreateFilePOptions.newBuilder().setRecursive(true).setWriteType(WritePType.THROUGH).build();
+    for (int dirId = 0; dirId < numDirs; dirId++) {
+      for (int fileId = 0; fileId < numFiles; fileId++) {
+        String filePath =
+            String.format("%s/%s/%d/%d", mParameters.mBasePath, mBaseParameters.mId, dirId, fileId);
+        createByteFile(fs, new AlluxioURI(filePath), options, fileSize);
+      }
+    }
+  }
+
+  private void createByteFile(FileSystem fs, AlluxioURI fileURI, CreateFilePOptions options,
+      int len) throws IOException, AlluxioException {
+    try (FileOutStream os = fs.createFile(fileURI, options)) {
+      byte[] arr = new byte[len];
+      for (int k = 0; k < len; k++) {
+        arr[k] = (byte) k;
+      }
+      os.write(arr);
+    }
+  }
+
+  private void deleteBasePath(FileSystem fs) throws IOException, AlluxioException {
+    AlluxioURI path = new AlluxioURI(mParameters.mBasePath);
+    if (fs.exists(path)) {
+      DeletePOptions options = DeletePOptions.newBuilder().setRecursive(true).build();
+      fs.delete(path, options);
+    }
+  }
+
+  @Override
+  public String getBenchDescription() {
+    return "";
+  }
+
+  @Override
+  public JobServiceBenchTaskResult runLocal() throws Exception {
+    ExecutorService service =
+        ExecutorServiceFactories.fixedThreadPool("bench-thread", mParameters.mNumDirs).create();
```

Comment:
What would happen if mNumDirs is very large?

Reply:
I mean, each job worker will run this and creates a thread pool of mNumDirs number of threads, and each thread will issue a distributedLoad targeting one directory, right? If there are thousands of directories or more, I feel like the job worker just can't create this many of threads?

Reply:
yeah I know what you mean, we have to be careful about the parameter for numDirs so it doesn't overload the job worker. And I feel like the test case we want to focus on is sending some concurrent requests(relative small) with lots of files per request. So I would not worry too much about this problem right now.